src/nodes/html.ts

```diff
@@ -1012,88 +1018,109 @@ export function base_parse(data: string, options = { lowerCaseTagName: false, co
 			return it.test(tag);
 		});
 	}
-	const root = new HTMLElement(null, {}, '', null);
```

Comment:
I cleaned up the logic a bit, added some comments, and used labels which made it more obvious what's happening. Some efficiency is gained, also.